typing/typeclass.ml

```diff
@@ -284,11 +289,12 @@ let enter_val cl_num vars inh lab mut virt ty val_env met_env par_env loc =
                      Field_type_mismatch("instance variable", lab, tr)))
     | Not_found -> None, virt
   in
-  let (id, _, _, _) as result =
-    match id with Some id -> (id, val_env, met_env, par_env)
+  let class_env = {val_env; met_env; par_env} in
```

Comment:
I don't think this line is necessary given that none of the `_env` variables have changed since they were introduced in scope -- `class_env` is the same value as the input parameter.

Reply:
(Compare to `enter_met_env` where the environment variables do change by shadowing.)

Reply:
Yes you are correct, I missed that.